DetectorDescription/Core/src/DDCompactView.cc

```diff
@@ -63,6 +64,15 @@ const DDLogicalPart& DDCompactView::root() const { return rep_->root(); }
 
 const DDPosData* DDCompactView::worldPosition() const { return worldpos_.get(); }
 
+const std::vector<double>& DDCompactView::vector(std::string_view iKey) const {
+  auto itFind = vectors_.find(std::string(iKey));
```

Comment:
Just curious, why not
```suggestion
const std::vector<double>& DDCompactView::vector(std::string const& iKey) const {
  auto itFind = vectors_.find(iKey);
```
?

Reply:
I was hoping in C++20 I could drop the string copy :)

Reply:
Thanks, sounds good.

Reply:
See https://en.cppreference.com/w/cpp/container/map/find which shows in C++14 they added the ability to do a find without having to create a key. Not sure why it didn't work here :(.

Reply:
https://stackoverflow.com/questions/35525777/use-of-string-view-for-map-lookup
seems to imply it might not be work it :(

Reply:
I see that for `std::unordered_map` the ability to allow `std::string_view` in `find` is not available until C++20.

https://en.cppreference.com/w/cpp/container/unordered_map/find

Reply:
> > The modified regression test runs fine. The vector registration by name is unsafe, but it is what we have now in the DD. The DDVectorGetter::beginWith is not used and can be dropped.
> 
> @ianna thanks so much for running the regression test. Was the change just the removal of the lines you pointed to earlier or is there some sort of reference output that also needs to be changed?

Yes, removing lines from 0084 to 0088 and also collecting the `DDVector` names just after https://cmssdt.cern.ch/lxr/source/DetectorDescription/RegressionTest/test/const_dump.cpp#0074

then iterating over the map from `DDCompactView`:
```c++
   for(const auto& it : vnames) {
      std::cout << it << std::endl;
      auto v = cpv.vector(it);
      std::cout << "size: " << v.size() << std::endl;
      for (double i : v) {
        std::cout << i << "\t";
      }
      std::cout << std::endl;
    }
```
Note, line 0074 returns `namespace:name`
```c++ 
vit->toString()
```

Reply:
@ianna Does the `cpv.vector(it)` actually work? As far as I can tell, `lockdown` is never called on that `DDCompactView` so the internal map is not filled.

Reply:
oh, yes... I've added the `lockdown` as well :-)

Reply:
after the parsing is done

Reply:
In the code snippet above, where did you get `vnames`?

Reply:
```diff
@@ -59,12 +59,15 @@ int main(int argc, char* argv[]) {
 
     DDErrorDetection ed(cpv);
     ed.report(cpv, std::cout);
+    cpv.lockdown();
 
     DDConstant::iterator<DDConstant> cit(DDConstant::begin()), ced(DDConstant::end());
     for (; cit != ced; ++cit) {
       cout << *cit << endl;
     }
 
+    std::vector<std::string> vnames;
+    
     DDVector::iterator<DDVector> vit;
     DDVector::iterator<DDVector> ved(DDVector::end());
     if (vit == ved)
@@ -72,6 +75,7 @@ int main(int argc, char* argv[]) {
     for (; vit != ved; ++vit) {
       if (vit->isDefined().second) {
         std::cout << vit->toString() << std::endl;
+       vnames.emplace_back(DDName(vit->name()).name());
         const std::vector<double>& tv = *vit;
         std::cout << "size: " << tv.size() << std::endl;
         for (double i : tv) {
@@ -80,12 +84,21 @@ int main(int argc, char* argv[]) {
         std::cout << std::endl;
       }
     }
-
-    std::vector<string> vnames;
-    DDVectorGetter::beginWith("Subdetector", vnames);
-    for (std::vector<string>::const_iterator sit = vnames.begin(); sit != vnames.end(); ++sit) {
-      std::cout << sit->c_str() << std::endl;
+    for(const auto& it : vnames) {
+      std::cout << it << std::endl;
+      auto v = cpv.vector(it);
+      std::cout << "size: " << v.size() << std::endl;
+      for (double i : v) {
+       std::cout << i << "\t";
+      }
+      std::cout << std::endl;
     }
+    

```

Reply:
[I should stop writing code before getting a response for stuff I ask others how they did it :)]